viewer/packages/pointclouds/wasm/src/parse_inputs.rs

```diff
@@ -0,0 +1,93 @@
+use nalgebra::Const;
+use std::vec::Vec;
+
+use crate::linalg::{vec3, Mat4, Vec3, Vec3WithIndex};
+use crate::shapes;
+
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+struct InputCylinder {
+    center_a: [f64; 3],
+    center_b: [f64; 3],
+    radius: f64,
+}
+
+#[derive(Debug, Deserialize)]
+struct InputOrientedBox {
+    inv_instance_matrix: [f64; 16],
+}
+
+#[derive(Debug, Deserialize)]
+pub struct InputShape {
+    object_id: u16,
+    cylinder: Option<Box<InputCylinder>>,
+    oriented_box: Option<Box<InputOrientedBox>>,
+}
```

Comment:
You're right, but I'm unable to make it work at the moment. In particular because we need to send in an array of complex structs, which is not supported by wasm-bindgen yet (https://github.com/rustwasm/wasm-bindgen/issues/111)